File: java/src/main/java/it/giannibombelli/wsc2026/payment/domain/payment/Provider.java
package it.giannibombelli.wsc2026.payment.domain.payment;

import it.giannibombelli.wsc2026.common.utils.Require;

public enum Provider {
    PAYPAL("PayPal"),
    KLARNA("Klarna"),
    GIFT_CARD("GiftCard");

    private final String label;

    Provider(String label) {
        this.label = label;
    }

    public String label() {
        return label;
    }

    public static Provider fromLabel(String label) {
        Require.requireArgument(label, "provider label");
        for (Provider provider : values()) {
            if (provider.label.equals(label)) {
                return provider;
            }
        }
        throw new IllegalArgumentException("unknown provider: " + label);
    }
}

File: typescript/src/payment/domain/payment/provider.ts
import { requireArgument } from '@/common/utils/requireArgument.js';
// Provider di pagamento supportati.

export const Provider = {
  PAYPAL: 'PayPal',
  KLARNA: 'Klarna',
  GIFT_CARD: 'GiftCard',
} as const;

export type Provider = (typeof Provider)[keyof typeof Provider];

export function providerFromLabel(label: string): Provider {
  requireArgument(label, 'provider label');
  const value = Object.values(Provider).find((provider) => provider === label);
  if (value === undefined) {
    throw new Error(`unknown provider: ${label}`);
  }
  return value;
}
